pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -112,7 +119,7 @@ func newClaimInfo(driverName, className string, claimUID types.UID, claimName, n
 }
 
 // newClaimInfoFromResourceClaim creates a new ClaimInfo object
-func newClaimInfoFromResourceClaim(resourceClaim *resourcev1alpha2.ResourceClaim) *ClaimInfo {
+func newClaimInfoFromResourceClaim(resourceClaim *resourcev1alpha2.ResourceClaim) ClaimInfo {
```

Comment:
Why this change? Normally "new" calls return a pointer.

Reply:
This is done to make this similar to `getClaimInfo`, which returns ClaimInfo value.

Reply:
I would still argue to return a pointer here for consistency, but I don't feel too strongly given how it's used in this component to assign to a variable that either gets its value from this function or one that (necessarily) returns a non-pointer type.

Reply:
I don't feel too strong about it either. I just thought that this code:
```
claimInfoCopy, err := m.cache.getClaimInfo(*claimName, pod.Namespace)
if err != nil {
	// claim does not exist in cache, create new claimInfo object
	// to be processed later.
	claimInfoCopy = newClaimInfoFromResourceClaim(resourceClaim)
}
```
would look a little bit more consistent if both `get` and `newClaimInfoFromResourceClaim` functions would return a value.

Reply:
one more point: if `newClaimInfoFromResourceClaim` we should probably test it for nil or assume it never returns nil.

Reply:
Let's leave it as a non-pointer